libsolidity/codegen/CompilerContext.cpp

```diff
@@ -265,31 +265,38 @@ void CompilerContext::appendInlineAssembly(
 	}
 
 	unsigned startStackHeight = stackHeight();
-	auto identifierAccess = [&](
+
+	assembly::ExternalIdentifierAccess identifierAccess;
+	identifierAccess.resolve = [&](
+		assembly::Identifier const& _identifier,
+		assembly::IdentifierContext
+	) {
+		auto it = std::find(_localVariables.begin(), _localVariables.end(), _identifier.name);
+		return it == _localVariables.end() ? size_t(-1) : 1;
+	};
+	identifierAccess.generateCode = [&](
 		assembly::Identifier const& _identifier,
-		eth::Assembly& _assembly,
-		assembly::CodeGenerator::IdentifierContext _context
+		assembly::IdentifierContext _context,
+		eth::Assembly& _assembly
 	) {
 		auto it = std::find(_localVariables.begin(), _localVariables.end(), _identifier.name);
-		if (it == _localVariables.end())
-			return false;
+		solAssert(it != _localVariables.end(), "");
 		unsigned stackDepth = _localVariables.end() - it;
 		int stackDiff = _assembly.deposit() - startStackHeight + stackDepth;
```

Comment:
Assuming there are only two kinds of variables,
* let declared variables
* other variables

I find it very strange that a check `m_context.isLocalVariable(variable)` appears.  I guess the there are more than two kinds of variables.

* [ ] I need to identify how many kinds of variables are assumed.
* [ ] I need to read the code again, making sure every kind of variables are treated in a reasonable way